ethcore/src/builtin.rs

```diff
@@ -393,11 +397,109 @@ impl Impl for Bn128MulImpl {
 	}	
 }
 
+mod bn128_gen {
+	use bn::{AffineG1, AffineG2, Fq, Fq2, G1, G2, Gt, pairing};
+
+	lazy_static! {
+		pub static ref P1: G1 = G1::from(AffineG1::new(
+			Fq::from_str("1").expect("1 is a valid field element"),
+			Fq::from_str("2").expect("2 is a valid field element"),
+		).expect("Generator P1(1, 2) is a valid curve point"));
+	}
+
+	lazy_static! {
```

Comment:
I believe these 3 `lazy_static` invocations could be combined. These are generators for `alt_bn128`, right? Eventually it would be nice to have these extracted to constants in a `bn::alt_gen` module, but not a blocker for now.